lib/debug_socket.rb

```diff
@@ -16,7 +17,10 @@ def self.logger
   end
 
   def self.start(path)
-    raise "debug socket thread already running" if @thread
+    pid = Process.pid
+    raise "debug socket thread already running for this process" if @thread && @pid == pid
```

Comment:
Yes, if you do infinite forks, you could open infinite debug sockets. At least that is what I was trying to do. I might have messed up the logic though.